src/e3/anod/qualifiers_manager.py

```diff
@@ -0,0 +1,799 @@
+            repr_alias=repr_alias,
+            repr_in_hash=repr_in_hash,
+        )
+
+        # Cleanly build the dictionary for mypy
+        new_tag_qualifier: QualifierTagDeclaration = {
+            "description": new_qualifier["description"],
+            "test_only": new_qualifier["test_only"],
+            "repr_alias": new_qualifier["repr_alias"],
+            "repr_in_hash": new_qualifier["repr_in_hash"],
+            "type": self.__TAG_QUALIFIER,
+        }
+
+        self.qualifier_decls[name] = new_tag_qualifier
+
+    def declare_key_value_qualifier(
+        self,
+        name: str,
+        description: str,
+        test_only: bool = False,
+        default: str | None = None,
+        choices: list[str] | None = None,
+        repr_alias: str | None = None,
+        repr_in_hash: bool = False,
+        repr_omit_key: bool = False,
+    ) -> None:
+        """Declare a new key value qualifier.
+
+        From the user prospective, 'declare' mean that a qualifier can be used in the
+        spec and will have an impact on the build space name.
+
+        From an internal point of view, this mean that the qualifier is added to
+        the qualifier_decls instance variable.
+
+        A key value qualifier is a 'standard' qualifier. They require the user to
+        provide their value.
+
+        This method cannot be called after the end of the declaration phase.
+
+        :param name: The name of the qualifier. It used to identify it and pass it to
+            the spec.
+        :param description: A description of the qualifier purposes. It is used to
+            make the help/error clearer.
+        :param test_only: By default the qualifier are used by all anod actions
+            (install, build, test...). If test_only is True, then this qualifier is
+            only available for test.
+        :param default: The default value given to the qualifier if no value was
+            provided by the user. If no default value is set, then the user must
+            provide a qualifier value at runtime.
+        :param choices: The list of all authorized values for the qualifier.
+        :param repr_alias: An alias for the qualifier name used by the name generation.
+            By default, the repr_alias is the qualifier name itself.
+        :param repr_in_hash: False by default. If True, the qualifier is included in
+            the hash at the end of the generated name. The result is less readable but
+            shorter.
+        :param repr_omit_key: If True, then the name generation don't display the
+            qualifier name/alias. It only use its value.
+        """
+        new_qualifier = self.__init_qualifier(
+            name=name,
+            description=description,
+            test_only=test_only,
+            repr_alias=repr_alias,
+            repr_in_hash=repr_in_hash,
+        )
+
+        # Cleanly build the dictionary for mypy
+        new_key_value_qualifier: QualifierKeyValueDeclaration = {
```

Comment:
I think it is better to directly create the object you need instead of creating a temp object and then changing it. What you are doing makes the code harder to read IMHO